lib/LedgerSMB/PGDate.pm

```diff
@@ -87,7 +87,7 @@ our $formats = {
 # Since we only match a limited set of patterns, below is what I
 # had expected DateTime::Format::Strptime would have done.
 my $regexes = {
-    'YYYY-MM-DD' => [ { regex => qr/^(\d{4,4})\-(\d\d)\-(\d\d)$/,
+    'YYYY-MM-DD' => [ { regex => qr/^(\d{4,4})\-(\d\d)\-(\d\d)\s?/,
```

Comment:
Hmm. This matches a space character or nothing. However, what when the input is 2016-12-2016 ? That's invalid input and the pattern was meant to reject the input.

Along the same lines, we don't support parsing a full iso 8601 DateTime at the moment, do we? I think that it'd make more sense to add an additional pattern which matches a full date+time for iso 8601